inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/region_yolo/region_yolo_kernel_ref.cpp

```diff
@@ -50,10 +50,17 @@ RegionYoloKernelRef::DispatchData SetDefault(const region_yolo_params& params) {
     RegionYoloKernelRef::DispatchData dispatchData;
 
     const auto& input = params.inputs[0];
-    if (input.GetLayout() == DataLayout::bfyx) {
-        dispatchData.gws = {input.X().v * input.Y().v, 1, 1};
-    } else {
+
+    switch (input.GetLayout())
+    {
+    case DataLayout::bfyx:
+    case DataLayout::byxf: {
+        uint32_t region_num = params.do_softmax ? params.num : params.mask_size;
+        dispatchData.gws = {input.X().v * input.Y().v, region_num, input.Batch().v};
+    } break;
+    default:
         dispatchData.gws = {input.Feature().v * input.Batch().v, input.X().v, input.Y().v};
+        break;
```

Comment:
doesnt this matter to be differ from above cases?

Reply:
I think this layer does not support other layers.. We can just throw exception.

Reply:
then you'll need to add Validate() function w.r.t the layouts

Reply:
@yeonbok  Is it necessary? It is already declared to support bfyx and byxf only from _gpu.cpp file.

Reply:
AFAIK, the param key is used by kernel selector. 
So  as you mentioned, the kerenl will be selected basically  as the supported layouts. 
But the Validate() can be used for various purpose... 
Curretnly I cannot see such kind of optimization but what if there will be post optimize graph?  
.. However, region yolo kernel might have very low possibility to be applied with such  post graph optimization ..  but I think having Validate code in each kernel will be good for future usage.

Reply:
By `It is already declared to support bfyx and byxf`, I meant not param key but primitive spec. BTW, what do you mean by "post optimize graph"? Is it possible to change layout during post_optimization?

Reply:
1. First I was not sure that whether the optimizations are never change kernel for a cldnn node directly. So I checked again the entire flow, and it seems that whenever it chooses a new kernel implementation it should chkeck the implementation map of the node.  So as you mentioned the layout will be always either of the two types. 
2. However still I think it is good to have that condition in Validate function for all kernels, if it has to be applied for specific layout. Wihtout that, if we change the cldnn operation spec to be appliable for more layouts in the future, at that time we will need to add this Validate code in the kernel. Adding Validate function is not a big deal to implement, so it would be nice to have always.

Reply:
Oh, I agree with your point.